src/enums/index.ts

```diff
@@ -1,138 +1,153 @@
-import { makeClass } from "./../utils/make-class";
-const assert = require("assert");
-const _ = require("lodash");
-const { parseBytes, serializeUIntN } = require("./../utils/bytes-utils");
-const enums = require("./definitions.json");
-
-function transformWith(func, obj) {
-  return _.transform(obj, func);
+import { serializeUIntN } from "./../utils/bytes-utils";
+import * as enums from "./definitions.json";
+
+const TYPE_WIDTH = 2;
+const LEDGER_ENTRY_WIDTH = 2;
+const TXN_TYPE_WIDTH = 2;
+const TXN_RESULT_WIDTH = 1;
+
+class EnumType {
+  bytes: Uint8Array;
+
+  constructor(
```

Comment:
Could `name` and `ordinalWidth` be enum values? Rather than untyped strings?

Reply:
ordinalWidth, yes. Not as sure about name, because we have to be able to go bidirectionally. Though it might make it easier, because the way values defined in `definitions.json` might make it easy.

(also because I'm not exactly sure how enums work in JS) I'll look into it and we can chat about it tomorrow.